tools/mlir-opt/CMakeLists.txt

```diff
@@ -61,3 +61,14 @@ add_llvm_executable(mlir-opt
 llvm_update_compile_flags(mlir-opt)
 whole_archive_link(mlir-opt ${LIBS})
 target_link_libraries(mlir-opt PRIVATE MLIRIR MLIRMlirOptLib ${LIBS} LLVMSupport)
+
+install(TARGETS mlir-opt
+  ${export_to_llvmexports}
+  COMPONENT mlir-opt
+  RUNTIME DESTINATION ${LLVM_TOOLS_INSTALL_DIR})
+
+if(NOT LLVM_ENABLE_IDE)
+  add_llvm_install_targets("install-mlir-opt"
+    DEPENDS mlir-opt
+  COMPONENT mlir-opt)
+endif()
```

Comment:
The LLVM tools don't need this, is this because they are using the `add_llvm_tool` macro instead of add_llvm_executable? Can we align?

This repeating pattern seems like it would be better behind a macro instead of spread all around.